R/brace_linter.R

```diff
@@ -124,7 +132,25 @@ brace_linter <- function(allow_single_line = FALSE) {
   # TODO(#1103): if c_style_braces is TRUE, this needs to be @line2 + 1
   xp_else_same_line <- glue("//ELSE[{xp_else_closed_curly} and @line1 != {xp_else_closed_curly}/@line2]")
 
-  xp_function_brace <- "(//FUNCTION | //OP-LAMBDA)/parent::expr[@line1 != @line2 and not(expr[OP-LEFT-BRACE])]"
+  if (function_braces != "never") {
+    xp_cond_function_brace <- switch(
+      function_braces,
+      always = "true",
+      multi_line = "@line1 != @line2",
+      not_inline = "@line1 != expr/@line1"
+    )
+
+    xp_function_brace <- glue(
+      "(//FUNCTION | //OP-LAMBDA)/parent::expr[{xp_cond_function_brace} and not(expr[OP-LEFT-BRACE])]"
+    )
+
+    msg_function_brace <- switch(
+      function_braces,
+      always = "Any function body should be wrapped in curly braces.",
+      multi_line = "Any function body spanning multiple lines should be wrapped in curly braces.",
+      not_inline = "Any function body starting on a new line should be wrapped in curly braces."
```

Comment:
```suggestion
      always = "Wrap function bodies in curly braces.",
      multi_line = "Wrap multi-line function bodies in curly braces.",
      not_inline = "Wrap function bodies starting on a new line in curly braces."
```

(will require updates to tests)